airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -106,6 +106,10 @@ def read(
                         f"The requested stream {configured_stream.stream.name} was not found in the source."
                         f" Available streams: {stream_instances.keys()}"
                     )
+                stream_is_available, error = stream_instance.check_availability(logger, self)
+                if stream_is_available is False:
+                    logger.warning(f"Skipped syncing stream '{stream_instance.name}' because it was unavailable. Error: {error}")
+                    continue
```

Comment:
This logic, as well as the logic above it checking whether streams in the configured catalog are in the source's streams, is due to be moved to a `filter_unavailable_streams` method in a separate PR. See #20252